author: Andres Salomon <dilinger@debian.org>
description: work around https://github.com/llvm/llvm-project/issues/48582

Can be dropped once we're using clang-16. Without this, bookworm's clang-14 errors with:

/../third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc:537:28: error: reference to local binding 'grid_items' declared in enclosing function 'blink::NGGridLayoutAlgorithm::BuildGridSizingSubtree'
    for (auto& grid_item : grid_items) {
                           ^
../../third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc:519:10: note: 'grid_items' declared here
  auto& [grid_items, layout_data, subtree_size] =
         ^



--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.cc
@@ -516,8 +516,12 @@ wtf_size_t NGGridLayoutAlgorithm::BuildG
                                 row_auto_repetitions);
 
   bool has_nested_subgrid = false;
-  auto& [grid_items, layout_data, subtree_size] =
+  // stupid workaround for https://github.com/llvm/llvm-project/issues/48582
+  auto& [g_i, l_d, s_s] =
       sizing_tree->CreateSizingData();
+  auto& grid_items = g_i;
+  auto& layout_data = l_d;
+  auto& subtree_size = s_s;
 
   if (!must_ignore_children) {
     // Construct grid items that are not subgridded.
@@ -1540,8 +1544,12 @@ void NGGridLayoutAlgorithm::InitializeTr
     NGGridSizingTree* sizing_tree) const {
   DCHECK(sizing_tree && current_grid_index < sizing_tree->Size());
 
-  auto& [grid_items, layout_data, subtree_size] =
+  // stupid workaround for https://github.com/llvm/llvm-project/issues/48582
+  auto& [g_i, l_d, s_s] =
       sizing_tree->At(current_grid_index);
+  auto& grid_items = g_i;
+  auto& layout_data = l_d;
+  auto& subtree_size = s_s;
 
   auto InitAndCacheTrackSizes = [&](GridTrackSizingDirection track_direction) {
     InitializeTrackCollection(opt_subgrid_data, track_direction, &layout_data);
